misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/Main.java

```diff
@@ -18,63 +18,119 @@
-            startTestSuit(Paths.get(response.getSourceRootPath()), response, jsonTmpSummaryPath);
+        Path testCache = Paths.get(args[0]);
+        String target = args[1];
+        boolean report = Boolean.valueOf(args[2]);
+        boolean coverage = Boolean.valueOf(args[3]);
+
+        if (report || coverage) {
+            testReport = new TestReport();
         }
-    }
 
-    private static String filterModuleName(String argument) {
-        //The module argument is always wrapped with a `\"` at start and end
-        if (argument.startsWith("\"") && argument.endsWith("\"") && argument.length() >= 2) {
-            return argument.substring(1, argument.length() - 1);
+        out.println();
+        out.print("Running Tests");
+        if (coverage) {
+            out.print(" with Coverage");
         }
-        return argument;
+        out.println();
+
+        Path testSuiteCachePath = testCache.resolve(TesterinaConstants.TESTERINA_TEST_SUITE);
+
+        try (BufferedReader br = Files.newBufferedReader(testSuiteCachePath, StandardCharsets.UTF_8)) {
+            Gson gson = new Gson();
+            Map<String, TestSuite> testSuiteMap = gson.fromJson(br,
+                    new TypeToken<Map<String, TestSuite>>() { }.getType());
+
+            if (!testSuiteMap.isEmpty()) {
+                for (Map.Entry<String, TestSuite> entry : testSuiteMap.entrySet()) {
+                    String moduleName = entry.getKey();
+                    TestSuite testSuite = entry.getValue();
+
+                    List<String> configList = new ArrayList<>();
+                    configList.add(target);
+                    configList.add(testSuite.getOrgName());
+                    configList.add(testSuite.getPackageName());
+                    configList.add("\"" + moduleName + "\"");
+                    configList.addAll(Lists.of(Arrays.copyOfRange(args, 4, args.length)));
+                    String[] configArgs = configList.toArray(new String[0]);
+
+                    LaunchUtils.initConfigurations(configArgs);
+
+                    out.println("\n\t" + (moduleName.equals(testSuite.getPackageName()) ?
+                            moduleName : testSuite.getPackageName() + TesterinaConstants.DOT + moduleName));
+
+                    testSuite.setModuleName(moduleName);
+                    List<String> testExecutionDependencies = testSuite.getTestExecutionDependencies();
+                    classLoader = createClassLoader(testExecutionDependencies);
+
+                    Path jsonTmpSummaryPath = testCache.resolve(moduleName).resolve(TesterinaConstants.STATUS_FILE);
+                    result = startTestSuit(Paths.get(testSuite.getSourceRootPath()), testSuite, jsonTmpSummaryPath,
+                            classLoader);
+                    exitStatus = (result == 1) ? result : exitStatus;
+                }
+            } else {
+                exitStatus = 1;
```

Comment:
Why do we set exit status as 1 when testSuiteMap is empty?
There can be cases where there are no tests right? We shouldn't mark such cases with with exit status 1 IMO. WDYT?

Reply:
The testSuiteMap contains 'TestSuites' for each module. If the map is empty, then the TestRunner shouldn't be running at all since no TestSuites have been initialized at compile time.

Even if the TestSuite has no tests, it will still count as part of the map. So having the map empty and running the TestRunner is the problem which is why the exit status is 1 in this case.